sdk/mobile-center/src/main/java/com/microsoft/azure/mobile/utils/async/DefaultSimpleFuture.java

```diff
@@ -0,0 +1,96 @@
+package com.microsoft.azure.mobile.utils.async;
+
+import com.microsoft.azure.mobile.utils.HandlerUtils;
+
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Implementation of {@link SimpleFuture}.
+ *
+ * @param <T> result type.
+ */
+public class DefaultSimpleFuture<T> implements SimpleFuture<T> {
+
+    /**
+     * Lock used to wait or monitor result.
+     */
+    private final CountDownLatch mLatch = new CountDownLatch(1);
+
+    /**
+     * Result
+     */
+    private T mResult;
+
+    /**
+     * Callbacks from thenAccept waiting for result
+     */
+    private Collection<SimpleConsumer<T>> mFunctions;
+
+    @Override
+    public T get() {
+        while (true) {
+            try {
+                mLatch.await();
+                break;
+            } catch (InterruptedException ignored) {
+            }
+        }
+        return mResult;
+    }
+
+    @Override
+    public boolean isDone() {
+        while (true) {
+            try {
+                return mLatch.await(0, TimeUnit.MILLISECONDS);
+            } catch (InterruptedException ignored) {
+            }
+        }
+    }
+
+    @Override
+    public synchronized void thenAccept(final SimpleConsumer<T> function) {
+        if (isDone()) {
+            HandlerUtils.runOnUiThread(new Runnable() {
```

Comment:
(And also I need to call back after the method returns the future, we already talked about that in the past for listener some time ago).